utils/kubernetes/describe/describe_test.go

```diff
@@ -0,0 +1,118 @@
+package describe
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	meshkitkube "github.com/layer5io/meshkit/utils/kubernetes"
+)
+
+type MockClient struct {
+	Object       string
+	Method       string
+	RequestUrl   string
+	ResponseCode int
+	Response     string
+	Client       *meshkitkube.Client
+}
+
+func TestDescribe(t *testing.T) {
+	ctx := context.Background()
+	//set up mock client  to handle api request and return expected responses
+	mckpod := MockClient{
+		Object:       "Pod",
+		Method:       "GET",
+		RequestUrl:   fmt.Sprintf("%s/api/v1/namespaces/default/pods?limit=10&continue=", ctx),
+		ResponseCode: 200,
+		Response:     "Mock response for GET request to /api/v1/namespaces/default/pods?limit=10&continue=abcdefg",
+	}
+	mckDeployment := MockClient{
+		Object:       "Deployment",
+		Method:       "POST",
+		RequestUrl:   fmt.Sprintf("%s/apis/apps/v1/namespaces/default/deployments", ctx),
+		ResponseCode: 200,
+		Response:     "Mock response for POST request to /apis/apps/v1/namespaces/default/deployments",
+	}
+	mckJob := MockClient{
+		Object:       "Job",
+		Method:       "GET",
+		RequestUrl:   fmt.Sprintf("%s/api/v1/job", ctx),
+		ResponseCode: 200,
+		Response:     "Moc Response for GET request to /api/v1/namespace",
+	}
+	mckcronjob := MockClient{
+		Object:       "Cronjob",
+		Method:       "PUT",
+		RequestUrl:   fmt.Sprintf("%s/apis/batch/v1beta1/namespaces/default/cronjobs", ctx),
+		ResponseCode: 200,
+		Response:     "Mock response for PUT request to /apis/batch/v1beta1/namespaces/default/cronjobs",
+	}
+	//create test cases
+	testCases := []struct {
+		Name          string
+		Options       DescriberOptions
+		ExpectedError error
+		clientApi     MockClient
+		Client        MockClient
+	}{
+		{
+			Name: "Pod",
+			Options: DescriberOptions{
+				Name:      "mypod",
+				Namespace: "test",
+				Type:      Pod,
+			},
+			ExpectedError: nil,
+			clientApi:     mckpod,
+		},
+		{
+			Name: "Deployment",
+			Options: DescriberOptions{
+				Name:      "deployment",
+				Namespace: "test",
+				Type:      Deployment,
+			},
+			ExpectedError: nil,
+			clientApi:     mckDeployment,
+		},
+		{
+			Name: "Job",
+			Options: DescriberOptions{
+				Name:      "job",
+				Namespace: "test",
+				Type:      Job,
+			},
+			ExpectedError: nil,
+			clientApi:     mckJob,
+		},
+		{
+			Name: "CronJob",
+			Options: DescriberOptions{
+				Name:      "cronjob",
+				Namespace: "test",
+				Type:      CronJob,
+			},
+			ExpectedError: nil,
+			clientApi:     mckcronjob,
+		},
+	}
+	// run test cases
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			// Register mock responders for each mock client
+			httpmock.RegisterResponder(tc.clientApi.Method, tc.clientApi.RequestUrl,
+				httpmock.NewStringResponder(tc.clientApi.ResponseCode, tc.clientApi.Response))
+
+			output, err := Describe(tc.Client.Client, tc.Options)
```

Comment:
The meshkitkube Client is not passed in any of the above test cases. You will have to initialise one and pass in each test case. It is initialised with a kubeconfig or when nothing is passed then it detects one from KUBECONFIG env variable. You can create a dummy kubeconfig and pass it to meshkitkube.New.